### PR TITLE
ci: add pre-commit with buildifier/prettier

### PR DESCRIPTION
**Pull request by JiaLiPassion**
Add pre-commit hooks with 
1. buildifier
2. commitizen
3. prettier

Also format current source.

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bazelrc
+**/*.md
```
Comment (jbedard) [inline .prettierignore:2]: Let's add the pnpm lockefile to this and revert there change to them

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# How to Contribute
+
+## Formatting
+
```
Comment (jbedard) [inline CONTRIBUTING.md:4]: If we're mentioning buildifier let's wish mention the is ones?

File: angular/projects/common/src/lib/common.component.ts [hunk 3]
```diff
@@ -2,11 +2,6 @@ import { Component } from '@angular/core';
 
 @Component({
   selector: 'lib-common',
-  template: `
-    <em>
-      common component!
-    </em>
-  `
+  template: ` <em> common component! </em> `,
```
Comment (jbedard) [inline angular/projects/common/src/lib/common.component.ts:5]: Can we avoid this one somehow?

Comment (JiaLiPassion) [reply angular/projects/common/src/lib/common.component.ts:5]: It seems it is not supported by `prettier` for now. https://github.com/prettier/prettier/issues/5588

File: CONTRIBUTING.md (hunk 2 above)
Comment (JiaLiPassion) [reply CONTRIBUTING.md:4]: I added some doc for `prettier` too.

File: .prettierignore (hunk 1 above)
Comment (JiaLiPassion) [reply .prettierignore:2]: Got it, done.

Comment (jbedard) [reply .prettierignore]: Should we only ignore the lockfiles and not all yaml though?

**Review by jbedard — APPROVED** [review]

File: .prettierignore (hunk 1 above)
Comment (JiaLiPassion) [reply .prettierignore]: Sure, updated.

**Pull request merged**